platform/linux-generic/pktio/dpdk.c

```diff
@@ -1493,21 +1489,50 @@ static void prepare_rss_conf(pktio_entry_t *pktio_entry,
 static int dpdk_input_queues_config(pktio_entry_t *pktio_entry,
 				    const odp_pktin_queue_param_t *p)
 {
+	struct rte_eth_dev_info dev_info;
+	pkt_dpdk_t *pkt_dpdk = pkt_priv(pktio_entry);
 	odp_pktin_mode_t mode = pktio_entry->param.in_mode;
 	uint8_t lockless;
+	int ret;
 
 	prepare_rss_conf(pktio_entry, p);
 
 	/**
 	 * Scheduler synchronizes input queue polls. Only single thread
 	 * at a time polls a queue */
-	if (mode == ODP_PKTIN_MODE_SCHED ||
-	    p->op_mode == ODP_PKTIO_OP_MT_UNSAFE)
+	if (mode == ODP_PKTIN_MODE_SCHED || p->op_mode == ODP_PKTIO_OP_MT_UNSAFE)
 		lockless = 1;
 	else
 		lockless = 0;
 
-	pkt_priv(pktio_entry)->flags.lockless_rx = lockless;
+	pkt_dpdk->flags.lockless_rx = lockless;
+
+	ret  = rte_eth_dev_info_get(pkt_dpdk->port_id, &dev_info);
+	if (ret) {
+		_ODP_ERR("DPDK: rte_eth_dev_info_get() failed: %d\n", ret);
+		return -1;
+	}
+
+	/* Configure RX descriptors */
+	for (uint32_t i = 0; i  < p->num_queues; i++) {
+		uint16_t num_rx_desc = pkt_dpdk->opt.num_rx_desc_default;
+
+		if (mode == ODP_PKTIN_MODE_DIRECT && p->queue_size[i] != 0) {
+			num_rx_desc = p->queue_size[i];
+			/* Make sure correct alignment is used */
+			if (dev_info.rx_desc_lim.nb_align)
+				num_rx_desc = RTE_ALIGN_MUL_CEIL(num_rx_desc,
+								 dev_info.rx_desc_lim.nb_align);
+		}
+
+		if (num_rx_desc < dev_info.rx_desc_lim.nb_min ||
+		    num_rx_desc > dev_info.rx_desc_lim.nb_max ||
+		    num_rx_desc % dev_info.rx_desc_lim.nb_align) {
+			_ODP_ERR("DPDK: invalid number of RX descriptors\n");
+			return -1;
```

Comment:
For easier debug, this could print num_rx_desc and the index.

Reply:
Fixed in V4.